test/TestVectorFieldCalc.cc

```diff
@@ -0,0 +1,131 @@
+    auto getdata_callback = [](std::vector<float>& data, CARTA::ImageBounds& bounds, int smoothing_factor,
+                                CARTA::PolarizationType stokes_type, int& width, int& height) {
+        float value = stokes_test_values_map[stokes_type]; // seems that Stokes I is passed as Current
+        // std::cout << "DEBUG : getdata_callback stokes = " << stokes_type << " value = " << value << std::endl;
+
+        data.assign(256 * 256, value); // generating Stokes I tile 256x256 all values = 1
+        width = 256;
+        height = 256;
+
+        return true;
+    };
+
+    DimsInfo dims;
+    dims.width = 2048;
+    dims.height = 2048;
+    dims.depth = 1;
+    dims.num_channels = 1;
+    dims.num_stokes = 4;
+    int z_index = 2;
+    vectorfield.Calculate(callback, dims, getdata_callback);
+
+    // check if intensities are as expected:
+    for (auto message : actual_intensities) {
+        const float* float_data = reinterpret_cast<const float*>(message.intensity_tiles(0).image_data().data());
+        int float_size = message.intensity_tiles(0).image_data().size() / 4;
+        std::vector<float> actual_intensity_values(float_data, float_data + float_size);
+        EXPECT_THAT(actual_intensity_values, Each(FloatNear(expected_intensity, 1e-5)));
+        // std::cout << "TEST " << actual_intensity_values[0] << " float_size = " << float_size << std::endl;
+    }
+
+    // check if angles are as expected:
+    for (auto message : actual_angles) {
+        const float* float_data = reinterpret_cast<const float*>(message.angle_tiles(0).image_data().data());
+        int float_size = message.angle_tiles(0).image_data().size() / 4;
+        std::vector<float> actual_angle_values(float_data, float_data + float_size);
+        EXPECT_THAT(actual_angle_values, Each(FloatNear(expected_angle, 1e-5)));
+    }
+}
+
+// Instantiate the test suite with the desired enum values
+INSTANTIATE_TEST_SUITE_P(StokesTests, VectorFieldCalcParamTest,
+    ::testing::Values(TestParameters(SourceTestMessage(VectorFieldCalculator::CURRENT, VectorFieldCalculator::CURRENT), true, 1, 1),
+        TestParameters(SourceTestMessage(VectorFieldCalculator::CURRENT, VectorFieldCalculator::COMPUTED), true, 1,
+            ((float)(180.0 / M_PI) * std::atan2(4, 3) / 2)), // computed PA
+        TestParameters(SourceTestMessage(VectorFieldCalculator::COMPUTED, VectorFieldCalculator::CURRENT), true, sqrt(3 * 3 + 4 * 4),
+            1), // Q=3 and U=4 -> Computed I=Q^2 + U^2
+        TestParameters(SourceTestMessage(VectorFieldCalculator::COMPUTED, VectorFieldCalculator::COMPUTED), true, sqrt(3 * 3 + 4 * 4),
+            ((float)(180.0 / M_PI) * std::atan2(4, 3) / 2)), // computed PA and Stokes I (as above)
+        TestParameters(SourceTestMessage(VectorFieldCalculator::COMPUTED, VectorFieldCalculator::CURRENT, false, 0.1, 0.2), true,
+            sqrt(3 * 3 + 4 * 4), 1), // de-biasing with errors in Q and U
```

Comment:
If these error values don't produce a different answer to the computed intensity without debiasing, try making them bigger (or make the FloatNear error smaller!)